kahuna/public/js/search/index.js

```diff
@@ -129,7 +129,8 @@ search.config(['$stateProvider', '$urlMatcherFactoryProvider',
     });
 
     $stateProvider.state('search.results', {
-        url: 'search?{query:Query}&ids&since&nonFree&uploadedBy&until&orderBy',
+        url: 'search?{query:Query}&ids&since&nonFree&uploadedBy&until&orderBy' +
+             '&dateField&takenSince&takenUntil&modifiedSince&modifiedUntil',
```

Comment:
I know you can use the date type, although I haven't done it before. i.e. `{modifiedUntil:date}`.
That applies to `boolean`s and `int`s too.